server/channels/app/metrics.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/request"
+)
+
+func (a *App) RegisterPerformanceReport(rctx request.CTX, report *model.PerformanceReport) *model.AppError {
+	if a.Metrics() == nil {
+		return nil
+	}
+
+	for _, c := range report.Counters {
+		switch c.Metric {
+		case model.ClientMetricChannelVisited:
+			a.Metrics().IncrementClientChannelVisited(report.Labels["platform"], report.Labels["agent"], float64(c.Value))
+		default:
+			return model.NewAppError("RegisterPerformanceReport", "", nil, "", http.StatusNotFound)
```

Comment:
Do we want to actually error out when a client sends an unrecognized metric, or should we just skip it and move on? I could see us wanting to remove metrics in the future once they're no longer useful, so it would be nice if older web apps could still send them without causing errors

Reply:
Hmm you're right, maybe moving on would be better here. Will update accordingly.